Kickstarter-iOS/Views/Controllers/RewardsCollectionViewController.swift

```diff
@@ -74,18 +80,13 @@ final class RewardsCollectionViewController: UICollectionViewController {
 
     guard let layout = self.flowLayout else { return }
 
-    let sectionInsets = layout.sectionInset
-    let topBottomInsets = sectionInsets.top + sectionInsets.bottom
-    let collectionViewSize = self.collectionView.frame.size
+    self.updateHiddenScrollViewBoundsIfNeeded(for: layout)
+  }
 
-    let itemHeight = self.collectionView.contentSize.height - topBottomInsets
-    var itemWidth = collectionViewSize.width - sectionInsets.left - 2 * peekAmountInset
+  override func viewWillTransition(to size: CGSize, with coordinator: UIViewControllerTransitionCoordinator) {
+    super.viewWillTransition(to: size, with: coordinator)
 
-    if [.landscapeLeft, .landscapeRight].contains(UIDevice.current.orientation) {
-      itemWidth = collectionViewSize.width / 3 - sectionInsets.left - 2 * peekAmountInset
-    }
-
-    layout.itemSize = CGSize(width: itemWidth, height: itemHeight)
+    self.flowLayout?.invalidateLayout()
```

Comment:
I've noticed that no matter what the current horizontal offset rotating the device always resets it to 0 therefore scrolls back to the very first reward. There should be a way to either cache current page and upon rotation restore the scroll offset (maybe inside the coordinator completion handler)?

Reply:
I think I'd prefer to implement this separately as an enhancement. I'll create a card 👍